### PR TITLE
MeshModelClass

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/w3d/renderer/w3d_file.h [hunk 1]
```diff
@@ -557,3 +558,145 @@ struct W3dVertInfStruct
...
+struct W3dTextureInfoStruct
+{
+    uint16_t Attributes;
+    uint16_t AnimType;
+    uint32_t FrameCount;
+    float FrameRate;
+};
+
+// clang-format off
+#define W3DTEXTURE_PUBLISH					0x0001
```
Comment (tomsons26) [inline src/w3d/renderer/w3d_file.h]: do change these tabs to spaces tho

Comment (jonwil) [reply src/w3d/renderer/w3d_file.h]: Fixed.

File: src/w3d/renderer/meshmdl.cpp [hunk 2]
```diff
@@ -14,15 +14,1374 @@
...
+    Reset(0, 0, 0);
+    Ref_Ptr_Release(m_matInfo);
+
+    if (m_defMatDesc != nullptr) {
+        delete m_defMatDesc;
+    }
+    if (m_alternateMatDesc != nullptr) {
+        delete m_alternateMatDesc;
+    }
+    return;
```
Comment (tomsons26) [inline src/w3d/renderer/meshmdl.cpp]: redundant

File: src/w3d/renderer/meshmdl.cpp [hunk 3]
```diff
@@ -14,15 +14,1374 @@
...
+    m_matInfo->Reset();
+    m_defMatDesc->Reset(polycount, vertcount, passcount);
+
+    if (m_alternateMatDesc != nullptr) {
+        delete m_alternateMatDesc;
+        m_alternateMatDesc = nullptr;
+    }
+
+    m_curMatDesc = m_defMatDesc;
+    return;
```
Comment (tomsons26) [inline src/w3d/renderer/meshmdl.cpp]: redundant

File: src/w3d/renderer/meshmdl.cpp [hunk 4]
```diff
@@ -14,15 +14,1374 @@
...
+    }
+}
+
+unsigned int MeshModelClass::Get_Sort_Flags(int pass) const
+{
+    unsigned int flags = 0;
+    ShaderClass::StaticSortCategoryType scat;
+
+    if (Has_Shader_Array(pass)) {
+        for (int tri = 0; tri < m_curMatDesc->m_shaderArray[pass]->Get_Count(); tri++) {
```
Comment (tomsons26) [inline src/w3d/renderer/meshmdl.cpp:1141]: tri > i

File: src/w3d/renderer/meshmdl.cpp (hunk 2 above)
Comment (jonwil) [reply src/w3d/renderer/meshmdl.cpp]: Fixed.

File: src/w3d/renderer/meshmdl.cpp (hunk 3 above)
Comment (jonwil) [reply src/w3d/renderer/meshmdl.cpp]: Fixed.

File: src/w3d/renderer/meshmdl.cpp (hunk 4 above)
Comment (jonwil) [reply src/w3d/renderer/meshmdl.cpp:1141]: Fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**